### PR TITLE
i18n(ja): Update `astro-components.md` translation

### PR DESCRIPTION
**Pull request by mottox2**
#### What kind of changes does this PR include?

- Translated content

#### Description

This PR update ja translation for [these changes](https://github.com/withastro/docs/commits/main/src/pages/en/core-concepts/astro-components.md?since=2022-08-09T21:13:13.000Z) in `astro-components.md`.

Please labeled this PR with `hacktoberfest-accepted`.

### CODE REVIEW

File: src/pages/ja/core-concepts/astro-components.md [hunk 1]
```diff
@@ -153,17 +159,35 @@ const items = ["犬", "猫", "カモノハシ"];
...
 
-```astro "El"
+```astro title="src/components/ConditionalHtml.astro" "visible"
 ---
-// src/pages/index.astro
-const El = 'div'
+const visible = true;
 ---
-<El>こんにちは！</El> <!-- <div>こんにちは！</div> としてレンダリングされます -->
+{visible && <p>私を見て！</p>}
```
Comment (kyosuke) [inline src/pages/ja/core-concepts/astro-components.md:168]: このままでも良い気がしますが、改行が抜けていることに気づきました。

```suggestion
{visible && <p>私を見て！</p>}

```

**Review by kyosuke — APPROVED** [review]
レビューしました。翻訳はわかりやすく問題ありませんでした！
サンプルコードの改行が一つ抜けていたのに気づいたのでコメントしています。

**Comment by mottox2** [comment]
レビューありがとうございます！抜けていた改行を追加しました

**Comment by kyosuke** [comment]
ありがとうございます！ LGTMです！

**Review by delucis — COMMENTED** [review]
Hi @mottox2! I’m sorry I didn’t manage to merge your PRs very quickly. We recently made one additional change to this page:

https://github.com/withastro/docs/commit/b34822eb26f396fe6e6e2c08822122a0cf295766#diff-2cea51d918d3ae4fedfce864624f37591ca58c59140f2f60c55e53641e91a661

Would you be able to update this PR with that?

File: src/pages/ja/core-concepts/astro-components.md [hunk 2]
```diff
@@ -153,17 +159,36 @@ const items = ["犬", "猫", "カモノハシ"];
...
+---
+<Element>こんにちは！</Element> <!-- <div>こんにちは！</div> としてレンダリングされます -->
+<Component /> <!-- <MyComponent />としてレンダリングされます -->
+```
+
+動的なタグを使用する場合：
+
+- **変数名をキャピタライズする** 例えば、`element`ではなくて`Element`を利用します。そうでなければ、Astroは変数名をリテラルなHTMLタグとしてレンダリングします。
+
+- **ハイドレーションのディレクティブはサポートしません.** [`client:*` ハイドレーションディレクティブ](/en/core-concepts/framework-components/#hydrating-interactive-components)を利用する場合、Astroは本番用にバンドルするコンポーネントを知る必要がありますが、動的なタグパターンではこれが動作しません。
```
Comment (delucis) [inline src/pages/ja/core-concepts/astro-components.md:191]: Links should always be to the `/ja/` version:

```suggestion
- **ハイドレーションのディレクティブはサポートしません.** [`client:*` ハイドレーションディレクティブ](/ja/core-concepts/framework-components/#hydrating-interactive-components)を利用する場合、Astroは本番用にバンドルするコンポーネントを知る必要がありますが、動的なタグパターンではこれが動作しません。
```

File: src/pages/ja/core-concepts/astro-components.md [hunk 3]
```diff
@@ -470,9 +491,21 @@ Astroは、これらのJavaScriptクライアントサイドインポートを
...
 ```
 
+## HTMLコンポーネント
+
+Astroは`.html`ファイルをコンポーネントとしてインポートして使用したり、これらのファイルをページとして`src/pages`のサブディレクトリに設置することをサポートしています。フレームワークなしで構築された既存のサイトからコードを再利用したい場合や、コンポーネントに動的な機能がないことを確認したい場合はHTMLコンポーネントを利用するといいでしょう。
+
+HTMLコンポーネントは有効なHTMLしか含むことができず、そのためAstroコンポーネントの主要機能が制限されます
+- frontmatterやサーバーサイドのインポート、動的な記法をサポートしません
+- すべての`<script>`タグはバンドルされずに残され、`is:inline`を持つ場合と同じように扱われます。
+- [`public/`フォルダにあるアセット](/en/guides/images/#public)のみを参照できます。
```
Comment (delucis) [inline src/pages/ja/core-concepts/astro-components.md:501]: Same here:

```suggestion
- [`public/`フォルダにあるアセット](/ja/guides/images/#public)のみを参照できます。
```

File: src/pages/ja/core-concepts/astro-components.md [hunk 4]
```diff
@@ -470,9 +491,21 @@ Astroは、これらのJavaScriptクライアントサイドインポートを
...
+
+:::note
+HTMLコンポーネント内の[`<slot />`要素](/en/core-concepts/astro-components/#slots)はAstroコンポーネントと同様に機能します。[HTML Web Component のSlot](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot) 要素を使うためには,  `<slot>` 要素に`is:inline`を追加します。
```
Comment (delucis) [inline src/pages/ja/core-concepts/astro-components.md:504]: And we can update both our internal link and the MDN link here for the Japanese versions!

```suggestion
HTMLコンポーネント内の[`<slot />`要素](/ja/core-concepts/astro-components/#slots)はAstroコンポーネントと同様に機能します。[HTML Web Component のSlot](https://developer.mozilla.org/ja/docs/Web/HTML/Element/slot) 要素を使うためには,  `<slot>` 要素に`is:inline`を追加します。
```

**Review by delucis — COMMENTED** [review]
Aha, and just seeing these links we need to correct.

File: src/pages/ja/core-concepts/astro-components.md [hunk 5]
```diff
@@ -470,9 +522,21 @@ Astroは、これらのJavaScriptクライアントサイドインポートを
...
+
+Astroは`.html`ファイルをコンポーネントとしてインポートして使用したり、これらのファイルをページとして`src/pages`のサブディレクトリに設置することをサポートしています。フレームワークなしで構築された既存のサイトからコードを再利用したい場合や、コンポーネントに動的な機能がないことを確認したい場合はHTMLコンポーネントを利用するといいでしょう。
+
+HTMLコンポーネントは有効なHTMLしか含むことができず、そのためAstroコンポーネントの主要機能が制限されます
+- frontmatterやサーバーサイドのインポート、動的な記法をサポートしません
+- すべての`<script>`タグはバンドルされずに残され、`is:inline`を持つ場合と同じように扱われます。
+- [`public/`フォルダにあるアセット](/ja/guides/images/#public)のみを参照できます。
+
+:::note
+HTMLコンポーネント内の[`<slot />`要素](/ja/core-concepts/astro-components/#slots)はAstroコンポーネントと同様に機能します。[HTML Web Component のSlot](https://developer.mozilla.org/ja/docs/Web/HTML/Element/slot) 要素を使うためには,  `<slot>` 要素に`is:inline`を追加します。
```
Comment (delucis) [inline src/pages/ja/core-concepts/astro-components.md:535]: Oops, I didn’t notice that this `#slots` link also needed updating to match the translated heading: `#スロット`

```suggestion
HTMLコンポーネント内の[`<slot />`要素](/ja/core-concepts/astro-components/#スロット)はAstroコンポーネントと同様に機能します。[HTML Web Component のSlot](https://developer.mozilla.org/ja/docs/Web/HTML/Element/slot) 要素を使うためには,  `<slot>` 要素に`is:inline`を追加します。
```

**Review by delucis — COMMENTED** [review]
Thanks for the quick work adding that extra translation @mottox2!

One small fix needed but then we should be good to merge 🥳

**Comment by mottox2** [comment]
Thanks for the review!

File: src/pages/ja/core-concepts/astro-components.md [hunk 6]
```diff
@@ -153,17 +190,36 @@ const items = ["犬", "猫", "カモノハシ"];
...
+<Element>こんにちは！</Element> <!-- <div>こんにちは！</div> としてレンダリングされます -->
+<Component /> <!-- <MyComponent />としてレンダリングされます -->
 ```
-:::
+
+動的なタグを使用する場合：
+
+- **変数名をキャピタライズする** 例えば、`element`ではなくて`Element`を利用します。そうでなければ、Astroは変数名をリテラルなHTMLタグとしてレンダリングします。
+
+- **ハイドレーションのディレクティブはサポートしません.** [`client:*` ハイドレーションディレクティブ](/ja/core-concepts/framework-components/#hydrating-interactive-components)を利用する場合、Astroは本番用にバンドルするコンポーネントを知る必要がありますが、動的なタグパターンではこれが動作しません。
```
Comment (delucis) [inline src/pages/ja/core-concepts/astro-components.md:222]: Oops, one more link — not sure why I missed this before. I’ll commit this and get this merged 🥳 

```suggestion
- **ハイドレーションのディレクティブはサポートしません.** [`client:*` ハイドレーションディレクティブ](/ja/core-concepts/framework-components/#インタラクティブなコンポーネント)を利用する場合、Astroは本番用にバンドルするコンポーネントを知る必要がありますが、動的なタグパターンではこれが動作しません。
```

**Comment by delucis** [comment]
Thanks again for the translation @mottox2 🙌

**Pull request merged**